pkg/operator/controller/extension/care/health.go

```diff
@@ -0,0 +1,195 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package care
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"k8s.io/utils/clock"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	gardencore "github.com/gardener/gardener/pkg/apis/core"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	operatorv1alpha1 "github.com/gardener/gardener/pkg/apis/operator/v1alpha1"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	"github.com/gardener/gardener/pkg/utils/flow"
+	"github.com/gardener/gardener/pkg/utils/gardener"
+	healthchecker "github.com/gardener/gardener/pkg/utils/kubernetes/health/checker"
+)
+
+// health contains information needed to execute health checks for the extension.
+type health struct {
+	extension           *operatorv1alpha1.Extension
+	gardenNamespace     string
+	runtimeClient       client.Client
+	virtualClient       client.Client
+	clock               clock.Clock
+	conditionThresholds map[gardencorev1beta1.ConditionType]time.Duration
+	healthChecker       *healthchecker.HealthChecker
+}
+
+// NewHealth creates a new Health instance with the given parameters.
+func NewHealth(
+	extension *operatorv1alpha1.Extension,
+	runtimeClient client.Client,
+	virtualClient client.Client,
+	clock clock.Clock,
+	conditionThresholds map[gardencorev1beta1.ConditionType]time.Duration,
+	gardenNamespace string,
+) HealthCheck {
+	return &health{
+		extension:           extension,
+		gardenNamespace:     gardenNamespace,
+		runtimeClient:       runtimeClient,
+		virtualClient:       virtualClient,
+		clock:               clock,
+		conditionThresholds: conditionThresholds,
+		healthChecker:       healthchecker.NewHealthChecker(runtimeClient, clock, conditionThresholds, nil),
+	}
+}
+
+// Check conducts the health checks on all the given conditions.
+func (h *health) Check(ctx context.Context, conditions ExtensionConditions) []gardencorev1beta1.Condition {
+	var taskFns []flow.TaskFn
+
+	if conditions.controllerInstallationsHealthy != nil {
+		taskFns = append(taskFns, func(_ context.Context) error {
```

Comment:
Isn't it better to use the `ctx` variable that will be passed to the function from inside the `Parallel` call? 
```suggestion
		taskFns = append(taskFns, func(ctx context.Context) error {
```

Currently, if the context handling inside `flow.Parallel` changes, it might lead to unexpected behaviour.